File: solutions/0961_long-pressed-name/solution.py
from itertools import groupby
class Solution:
    def isLongPressedName(self, name: str, typed: str) -> bool:
        if set(name) != set(typed):
            return False
        g1 = [list(g) for k,g in groupby(name)]
        g2 = [list(g) for k,g in groupby(typed)]
        if len(g1) != len(g2):
            return False
        for i in range(len(g2)):
            c1 = g1[i][0]
            c2 = g2[i][0]
            if(c1!=c2):
                return False
            elif len(g1[i]) > len(g2[i]) :
                return False
        return True

File: solutions/0961_long-pressed-name/solution.cpp
class Solution {
public:
    bool isLongPressedName(string name, string typed) {
        vector <pair<char,int>> n;
        vector <pair<char,int>> t;
        char curr = name[0];
        int count = 0;
        for(int i = 0;i<name.size();i++){
            if(curr==name[i]){
                count++;
            }else{
                n.push_back({curr,count});
                curr = name[i];
                count = 1;
            }
        }
        n.push_back({curr,count});
        curr = typed[0];
        count = 0;
        for(int i = 0;i<typed.size();i++){
            if(curr==typed[i]){
                count++;
            }else{
                t.push_back({curr,count});
                curr = typed[i];
                count = 1;
            }
        }
        t.push_back({curr,count});
        if (n.size()!=t.size()){
            return false;
        }
        for(int j=0;j<n.size();j++){
            if(n[j].first != t[j].first){
                return false;
            }else if(n[j].second > t[j].second){
                return false;
            }
        }
        return true;
    }
};